src/metabase/permissions/models/data_permissions.clj

```diff
@@ -791,21 +791,28 @@
             (when (not= values #{existing-db-perm-value})
               ;; If we're setting any table permissions to a value that is different from the database-level permission,
               ;; we need to replace it with individual permission rows for every table in the database instead.
-              (let [other-tables    (t2/select :model/Table {:where [:and
-                                                                     [:= :db_id db-id]
-                                                                     [:not [:in :id table-ids]]]})
-                    other-new-perms (map (fn [table]
-                                           {:perm_type   perm-type
-                                            :group_id    group-id
-                                            :perm_value  (case existing-db-perm-value
-                                                           ;; If the previous database-level permission can't be set at
-                                                           ;; the table-level, we need to provide a new default
-                                                           :query-builder-and-native :query-builder
-                                                           existing-db-perm-value)
-                                            :db_id       db-id
-                                            :table_id    (:id table)
-                                            :schema_name (:schema table)})
-                                         other-tables)]
+              (let [other-new-perms (->> (t2/select :model/Table {:where
```

Comment:
might be worth breaking this out into a helper function at this point?

Reply:
maybe a unit test could make sense for this helper function as well? I know it's hard to repro the original issue in a test since it relies on large table counts, but maybe we could do something with synthetic table IDs to ensure that the filtering is working properly?